### PR TITLE
JaCoCo tool integrated in the project.

### LINKED ISSUE #69: Add JaCoCo coverage report locally.
I want to integrate the JaCoCo tool to check the coverage of our code. For now, we only want to do it locally. But in the future could be useful adding a new job in the pipeline to check the coverage.

### PR DESCRIPTION
**Pull request by RubenGonz2**
### :tophat: What is the goal?
Integrate the JaCoCo tool to get the testing coverage report for the project.

In this first step we will not add a new step in the pipeline to check the coverage.

Even if we have only one module, I have prepared the script to be ready to work with a multi-module project and merge all the source code and the result for the tests in all the modules of the project.

### :memo: Type of change
- [x] New feature (non-breaking change which adds functionality)

This closes 
- #69

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,7 +16,14 @@ plugins {
 }
 
 apply from: 'gradle-scripts/detekt.gradle'
+apply from: 'gradle-scripts/jacoco.gradle'
 
 task clean(type: Delete) {
     delete rootProject.buildDir
 }
+
+if (rootProject.ext.isLocalBuild) {
```
Comment (soygabimoreno) [inline build.gradle:25]: :heart:

File: documentation/jacoco.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# JaCoCo
+
+JaCoCo is an open-source toolkit for measuring and reporting Java code coverage.
+
+Of course, it's compatible for Kotlin code too. THat is very important on **`Los Androides`**.
```
Comment (soygabimoreno) [inline documentation/jacoco.md]: Nit: ```suggestion
Of course, it's compatible with Kotlin code too. That is very important on **`Los Androides`**.
```

File: documentation/jacoco.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Resources used to implement this solution
+
+- https://medium.com/nerd-for-tech/setup-jacoco-code-coverage-with-your-multimodule-android-app-kotlin-a0f82573a1
+- https://thsaravana.github.io/blog/jacoco-single-coverage-for-multi-module/
+
+# Relevant information int the JaCoCo Coverage Report
```
Comment (soygabimoreno) [inline documentation/jacoco.md]:
```suggestion
# Relevant information in the JaCoCo Coverage Report
```

File: gradle-scripts/jacoco.gradle [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                    }
+                }
+                returnDefaultValues = true
+            }
+
+            buildTypes.debug.testCoverageEnabled true
+        }
+    }
+
+
```
Comment (soygabimoreno) [inline gradle-scripts/jacoco.gradle]: Nit: Surplus empty line

**Review by soygabimoreno — APPROVED** [review]
Absolutely amazing, @RubenGonz2! 🏆️

File: documentation/jacoco.md (hunk 3 above)
Comment (RubenGonz2) [reply documentation/jacoco.md]: Fixed

File: gradle-scripts/jacoco.gradle (hunk 4 above)
Comment (RubenGonz2) [reply gradle-scripts/jacoco.gradle]: Fixed

**Review by joseantoniocampillo — APPROVED** [review]
Great! 😃👍

**Comment by RubenGonz2** [comment]
I have added the JaCoCo report in the pipeline, publishing the HTML report as an artifact, and add a comment automatically in the PR with the coverage for the files changed in the PR and the global coverage.

**Pull request merged**